### PR TITLE
Don't clash with X11 sockets used by the system

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Workaround for #60

### CODE REVIEW

**Comment by Saviq** [comment]
Can you please fold (a version of) #59 into this? I'd be +1 on bailing with an error if the interface is connected.

**Comment by Saviq** [comment]
Also, should this actually target `main` and we cherry-pick into `core20`?

**Comment by AlanGriffiths** [comment]
> Also, should this actually target `main` and we cherry-pick into `core20`?

done

File: performance/performance-env.sh [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
-  exec "$@" --gtest_filter=-GLMark2Xwayland.*
-fi
\ No newline at end of file
+  echo "WARNING: some of these tests (GLMark2Xwayland.*) fail with the x11 interface connected"
+
+  if ! echo "$@" | grep --quiet -e "\-\-gtest_filter=" -e "/bin/bash"; then
+    echo "ERROR:  Please either disconnect the x11 plug or provide a --gtest_filter:"
+    echo "    sudo snap disconnect ${SNAP_INSTANCE_NAME}:x11"
+    echo "  Or:"
+    echo "    ${SNAP_INSTANCE_NAME} --gtest_filter=-GLMark2Xwayland.*"
```
Comment (Saviq) [inline performance/performance-env.sh]:
```suggestion
    echo "    ${SNAP_INSTANCE_NAME}.performance-test --gtest_filter=-GLMark2Xwayland.*"
```

**Review by Saviq — APPROVED** [review]
Good otherwise!

bors delegate+

**Comment by AlanGriffiths** [comment]
> Good otherwise!

Good catch!

bors r=Saviq

**Pull request merged**